constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			externs := []string{"data.inventory"}
+			if tt.extern != nil {
+				externs = tt.extern
+			}
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, externs)
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			ref, err := ast.ParseRef(tt.imports)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			wantRef, err := ast.ParseRef(tt.want)
+			if err != nil {
+				t.Fatalf("Failed to parse ref %s", err)
+			}
+			if got := r.rewriteDataRef(ref); !got.Equal(wantRef) {
+				t.Errorf("rewriteDataRef() = %q, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAddEntryPointParseErrors(t *testing.T) {
+	tcs := []struct {
+		name      string
+		path      string
+		src       string
+		wantError error
+	}{
+		{
+			name: "add lib.rego path",
+			path: "data.stuff.foolib",
+			src: `package lib.rego
+
+test_ok {
+	true
+}
+`,
+		},
+		{
+			name: "add invalid rego",
+			path: "invalidrego",
+			src: `package lib.rego
+something invalid`,
+			wantError: ErrInvalidModule,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := MockRegoWriter()
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			if gotErr := rr.AddEntryPoint(tc.path, tc.src); !errors.Is(gotErr, tc.wantError) {
+				t.Errorf("got AddEntryPoint() error = %q, want %v", gotErr, tc.wantError)
+			}
+		})
+	}
+}
+
+func TestRegoRewriterAddPathFromFs(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr error
+	}{
+		{
+			name:    "path not exist",
+			path:    "foo/bar",
+			wantErr: ErrReadingFile,
+		},
+		{
+			name: "path with test folder",
+			path: ".",
+		},
+		{
+			name: "path without test",
+			path: "../../",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := MockRegoWriter()
+			if err != nil {
+				t.Fatalf("Failed to create %s", err)
+			}
+			if err := rr.AddLibFromFs(tc.path); !errors.Is(err, tc.wantErr) {
+				t.Errorf("AddLibFromFs() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if err := rr.AddBaseFromFs(tc.path); !errors.Is(err, tc.wantErr) {
+				t.Errorf("AddBaseFromFs() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestRegoRewriterEntryPoint(t *testing.T) {
+	tests := []struct {
+		name       string
+		prefix     string
+		content    string
+		wantResult map[string]string
+		wantError  error
+	}{
+		{
+			name:   "entry point imports lib",
+			prefix: "prefix.path",
+			content: `package lib.alpha
+				import data.lib.alpha
+        violation[{"msg":msg}] {
```

Comment:
The spacing for this is weird - a mix of spaces and tabs. Is that intentional and part of the test? If not, stick to tabs since `wantResult` uses tabs so it seems the rewriter normalizes to tabs.

Reply:
Removed the indent in test source.